.github/workflows/sycl-precommit.yml

```diff
@@ -69,27 +65,38 @@ jobs:
           else
             echo 'arc_tests="Matrix/"' >> "$GITHUB_OUTPUT"
           fi
-  test:
-    needs: [build, detect_changes, determine_arc_tests]
-    if: ${{ always() && !cancelled() && needs.build.outputs.build_conclusion == 'success' }}
+
+  test_linux:
+    needs: [build_linux, detect_changes, determine_arc_tests]
+    if: ${{ always() && !cancelled() && needs.build_linux.outputs.build_conclusion == 'success' }}
     strategy:
       fail-fast: false
       matrix:
         include:
-          - name: AMD/HIP
+          # XXX CUDA testing is in a separate workflow because it runs on AWS
+          # which has different access restrictions/requirements. See
+          # sycl-linux-precommit.yml if you're looking for the CUDA precommit
+          # definition
+          - name: E2E on AMD/HIP
             runner: '["Linux", "amdgpu"]'
             image: ghcr.io/intel/llvm/ubuntu2204_build:latest-0300ac924620a51f76c4929794637b82790f12ab
             image_options: -u 1001 --device=/dev/dri --device=/dev/kfd
             target_devices: ext_oneapi_hip:gpu
-          - name: Intel
+
+          - name: E2E on Intel GEN12; level zero GPU, OpenCL CPU
             runner: '["Linux", "gen12"]'
             image: ghcr.io/intel/llvm/ubuntu2204_intel_drivers:latest
-            image_options: -u 1001 --device=/dev/dri --privileged --cap-add SYS_ADMIN
             target_devices: ext_oneapi_level_zero:gpu;opencl:gpu;opencl:cpu
             reset_gpu: true
             install_drivers: ${{ contains(needs.detect_changes.outputs.filters, 'drivers') }}
             extra_lit_opts: --param gpu-intel-gen12=True
-          - name: E2E tests on Intel Arc A-Series Graphics
+
+          - name: E2E on Intel GEN12 Graphics; OpenCL GPU, OpenCL FPGA
+            runner: '["Linux", "gen12"]'
+            extra_lit_opts: --param gpu-intel-gen12=True
+            target_devices: opencl:gpu;opencl:fpga
```

Comment:
This might not be an issue introduced by this PR  and may already exist, but I think having `gpu-intel-gen12=True` and multiple target devices means that the variable is true for all target device (unless we somehow set it to false in the lit python scripts if it's not gpu). Probably we only want that variable to be true for GPU targets.

 @aelovikov-intel Any idea?

Reply:
There is a parallel activity to auto-detect architecture. I think that would solve the issue, but I don't remember the status of it.

Reply:
It should be already available: #13976

Reply:
> Cool, thanks! Probably we need to either map the results there to the `gpu-intel-*` we have been using, or the other way around. Having two ways to specific the gpu seems bad.

That's the "in progress" part that I'm unaware about the status of. @AlexeySachkov , someone from your team is looking into it, right?